test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
 		//Select test by name if --singletest is set and not filling state tests as blockchain
 		if (!Options::get().fillchain && !TestOutputHelper::passTest(testname))
 		{
-			o.clear(); //don't add irrelevant tests to the final file when filling
-			continue;
+			if (_fillin)
+			{
+				o.clear(); //don't add irrelevant tests to the final file when filling
+				continue;
+			}
+			else
+			{
+				//Select test case from file by combining singleTestName and SingleTestNet
+				//singleTestName selects the fillerfile and SingleTestNet select particular test case in result test file
+				if (Options::get().singleTestName + "_" + Options::get().singleTestNet != testname)
```

Comment:
the issue is that in blockchain filler test has  name  "test" 
but when blockchain test is generated it now has several tests with names
"test_EIP150" "test_Frontier"   and so on. 

so when you run filler you could select single test by  --singletest test
but when you run bctest you could selet single test by  --singletest test --singlenet Frontier